app/models/user.rb

```diff
@@ -89,4 +92,16 @@ def default_email_record
     result ||= valid_emails.first
     result
   end
+
+  # Overrides `Devise::Models::Validatable`
```

Comment:
I think grouping functionality is better -- that's essentially what concern means.

Overriding email_required? and password_required? is also an implicit coupling with Devise, so in this case I rather put all the Devise things together. The system? call is our own API so it's less of a problem.